From 4923e67d2a4a71d70785acc7cce28468684f301e Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 25 Jan 2016 01:02:02 +0700
Subject: [PATCH 04/64] Revert "Revert "Revert "NBV: inline trivial method" and
 the revert revert that RE-BROKE something.""

This reverts commit 118f2fd6cf7147c2b1f0104a9550061b756ea254.
---
 .../android/systemui/statusbar/phone/NavigationBarView.java    | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
index 4043c47..27229eb 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
@@ -495,7 +495,7 @@ public class NavigationBarView extends LinearLayout {
     final Runnable mNotifyLayoutChanged = new Runnable() {
         @Override
         public void run() {
-            setupNavigationButtons(getCurrentButtonArray());
+            loadButtonArrays();
         }
     };
 
@@ -691,7 +691,7 @@ public class NavigationBarView extends LinearLayout {
                     for(int i=0;i<mButtonLayouts;i++)
                         mButtonContainerStrings[i] = Settings.System.getString(r, buttonSettings[i]);
 
-                    loadButtonArrays();
+                    notifyLayoutChange(0);
                 }};
 
             for(int i=0;i<5;i++)
@@ -773,7 +773,7 @@ public class NavigationBarView extends LinearLayout {
                 mIMEKeyArray.add(new KeyButtonInfo(actions[0], actions[1], actions[2], actions[3]));
             }
         }
-        setupNavigationButtons(getCurrentButtonArray());
+        setupNavigationButtons();
     }
 
     private ArrayList<KeyButtonInfo> getButtonsArray(final String[] userButtons) {
@@ -790,6 +790,10 @@ public class NavigationBarView extends LinearLayout {
         return mAllButtonContainers.get(mCurrentLayout);
     }
 
+    private void setupNavigationButtons() {
+        setupNavigationButtons(getCurrentButtonArray());
+    }
+
     private void setupNavigationButtons(ArrayList<KeyButtonInfo> buttonsArray) {
         final boolean stockThreeButtonLayout = buttonsArray.size() == 3;
         final int separatorSize = (int) mMenuButtonWidth;
-- 
2.5.0

